sys/sema/sema.c

```diff
@@ -46,6 +49,7 @@ void sema_destroy(sema_t *sema)
     mutex_unlock(&sema->mutex);
 }
 
+#if !IS_USED(MODULE_SEMA_NO_XTIMER)
```

Comment:
I tried to look at how other of our API's take care of this issue:

- `sock`: timeout functionality will only work if `MODULE_XTIMER` is actually pulled in, but the API has no mention for this
- `mutex`, `msg`: timeout functionality is provided by the different timer abstraction modules and the API specifies that.
- `event`: if the modules are included then the timeout functions will be available, the API itself does not say this unless code is checked and an `ifdef` is seen.

IMO the second is the best approach for `core` to avoid timing dependencies. But since in this case it's a sys module, I think the third approach combined with some better API doc would be the best, for example, a `@pre xtimer/ztimer must be included`. A pseudo module could also be nice but not necessary, I would also rather have it as opt-in instead of opt-out, maybe `sema-timeout`. What do you think?